components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/AbstractAuthorizationGrantHandler.java

```diff
@@ -1026,6 +1028,15 @@ private void clearExistingTokenFromCache(OAuthTokenReqMessageContext tokenMsgCtx
                     OAuth2Util.buildScopeString(existingTokenBean.getScope()));
             OAuthUtil.clearOAuthCache(existingTokenBean.getConsumerKey(), existingTokenBean.getAuthzUser());
             OAuthUtil.clearOAuthCache(existingTokenBean);
+            try {
+                if (oauthTokenIssuer != null && OAuth2TokenUtil.isJWT(existingTokenBean.getAccessToken())) {
+                    String tokenAlias = oauthTokenIssuer.getAccessTokenHash(existingTokenBean.getAccessToken());
+                    OAuthUtil.clearOAuthCache(tokenAlias);
+                }
+            } catch (OAuthSystemException e) {
+                log.error("Error while clearing cache for the access token : " +
```

Comment:
better to throw the exception and handle it in upper lelve. You can catch and throw IdentityOAuth2Exception as already IdentityOAuth2Exception is thrown from the caller.